SetupBLT.cmake

```diff
@@ -66,6 +66,13 @@ if (NOT BLT_LOADED)
     ################################
     include(${BLT_ROOT_DIR}/cmake/BLTOptions.cmake)
 
+    ################################
+    # CMP0057 New is required by blt_setup_target()
+    # in order to support the IN_LIST if() operator
+    ################################
+    cmake_policy(SET CMP0057 NEW)
```

Comment:
I think this either does not belong here, or it must be guarded.  

If left as is, it will trigger an error in earlier versions of cmake.  See https://cmake.org/cmake/help/v3.12/manual/cmake-policies.7.html

Specifically, if we add it here, please change it to:
```cmake
if(POLICY CMP0057)
  cmake_policy(SET CMP0057 NEW)
endif()
```

Reply:
Ticket #109 is relevant.  

Perhaps we should move logic related to setting appropriate policies (based on the cmake version) into its own file.

Reply:
I think the more alarming situation will be when using cmake version that don't support CMP0057, any blt macro that calls blt_setup_target() using a list of dependencies will fail. For instance 
```
blt_add_library( 
NAME foo
SOURCES bar1.cpp bar2.cpp
)
```
I think that ticket #109 addresses this in more detail. Can we leave this here for now so that the CI tests can pass and wait for resolution on #109 ?

Reply:
That's fine, but I don't see the downside in guarding it as in my snippet above.  It shouldn't break any current tests since it will always set the policy when that policy is defined.

I don't think we've addressed the question of the minimum cmake version that blt supports.  Certain features require newer versions (e.g. for cuda support).

I think the ``IN_LIST`` thing will only bite users of older cmake versions when they use ``blt_add_library`` and the added library has dependencies.